core/policy/policy_repository.go

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package policy
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/mysteriumnetwork/node/market"
+	"github.com/mysteriumnetwork/node/requests"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+)
+
+type policyMetadata struct {
+	policy market.AccessPolicy
+	rules  market.AccessPolicyRuleSet
+}
+
+// PolicyRepository represents async policy fetcher from TrustOracle
+type PolicyRepository struct {
+	client *requests.HTTPClient
+
+	policyURL  string
+	policyLock sync.Mutex
+	policyList []policyMetadata
+
+	fetchInterval time.Duration
+	fetchShutdown chan struct{}
+}
+
+// NewPolicyRepository create instance of policy repository
+func NewPolicyRepository(client *requests.HTTPClient, policyURL string, interval time.Duration) *PolicyRepository {
+	return &PolicyRepository{
+		client:        client,
+		policyURL:     policyURL,
+		policyList:    make([]policyMetadata, 0),
+		fetchInterval: interval,
+		fetchShutdown: make(chan struct{}),
+	}
+}
+
+// Start begins fetching proposals to repository
+func (pr *PolicyRepository) Start() {
+	pr.fetchShutdown = make(chan struct{})
+	go pr.fetchLoop()
+}
+
+// Stop ends fetching proposals to repository
+func (pr *PolicyRepository) Stop() {
+	pr.fetchShutdown <- struct{}{}
```

Comment:
close it, since sending a single struct will fail to signal multiple listeners(if you ever have them)